src/main/java/dk/alexandra/fresco/lib/helper/builder/AbstractProtocolBuilder.java

```diff
@@ -146,7 +177,19 @@ public ProtocolProducer getProtocol() {
 	
 	@Override
 	public void reset() {
+		if(parent != null){
+			parent.reset();
+			return;
+		}
 		curpp = new SequentialProtocolProducer();
 		producerStack = new Stack<AppendableProtocolProducer>();
 	}
+	
+	/**
+	 * Sets the parent of this builder meaning that the internal stack of the builder will be the parents. 
+	 * @param parent
+	 */
+	protected void setParentBuilder(AbstractProtocolBuilder parent){
```

Comment:
This could be made public, granting the power of making any builder into a master builder. This would then support the style of coding that we currently have, while still avoiding problems with multiple builders.